api/src/routes/retrieve.js

```diff
@@ -5,6 +5,7 @@ import asyncRoute from '../utils/async-route.js';
 export default () => asyncRoute(async (req, res) => {
   const { query } = req;
   const { tenant, realm } = query;
+  const granularity = query.granularity ? query.granularity : 'week';
```

Comment:
You can do `const granularity = query.granularity || 'week'` here since a falsy query will fallback to 'week' via the or operator